pallets/transfer-allowlist/src/benchmarking.rs

```diff
@@ -0,0 +1,298 @@
+			Pallet::<T>::add_transfer_allowance(RawOrigin::Signed(sender.clone()).into(), CurrencyId::Native, receiver.clone().into())?;
+		  advance_n_blocks::<T>(1u32.into());
+  }:remove_transfer_allowance(RawOrigin::Signed(sender.clone()), CurrencyId::Native, receiver.clone().into())
+			verify {
+				  assert_eq!(
+						  Pallet::<T>::get_account_currency_transfer_allowance(
+								  (sender,
+								   CurrencyId::Native,
+								   Location::from(receiver))
+						  ).unwrap(),
+							AllowanceDetails {
+								  allowed_at: T::BlockNumber::one(),
+								  blocked_at: <frame_system::Pallet<T>>::block_number().checked_add(&delay).expect("Invalid blocked at."),
+							}
+					)
+			}
+
+	remove_transfer_allowance_no_delay {
+		  let (sender, receiver) = set_up_users::<T>();
+			Pallet::<T>::add_transfer_allowance(RawOrigin::Signed(sender.clone()).into(), CurrencyId::Native, receiver.clone().into())?;
+	}:remove_transfer_allowance(RawOrigin::Signed(sender.clone()), CurrencyId::Native, receiver.clone().into())
+		  verify {
+			  assert_eq!(
+					  Pallet::<T>::get_account_currency_transfer_allowance(
+							  (sender,
+							   CurrencyId::Native,
+							   Location::from(receiver))
+					  ).unwrap(),
+						AllowanceDetails {
+							  allowed_at: T::BlockNumber::zero(),
+							  blocked_at: <frame_system::Pallet<T>>::block_number(),
+						}
+				)
+		  }
+
+	purge_transfer_allowance_no_remaining_metadata {
+			let (sender, receiver) = set_up_users::<T>();
+			  Pallet::<T>::add_transfer_allowance(RawOrigin::Signed(sender.clone()).into(), CurrencyId::Native, receiver.clone().into())?;
+			Pallet::<T>::remove_transfer_allowance(RawOrigin::Signed(sender.clone()).into(), CurrencyId::Native, receiver.clone().into())?;
+	}:purge_transfer_allowance(RawOrigin::Signed(sender.clone()), CurrencyId::Native, receiver.clone().into())
+	verify {
+			  assert_eq!(
+					  Pallet::<T>::get_account_currency_transfer_allowance(
+							  (sender.clone(),
+							   CurrencyId::Native,
+							   Location::from(receiver))
+					  ),
+			None
+				);
+			assert_eq!(
+
+					Pallet::<T>::get_account_currency_restriction_count_delay(
+							sender,
+							CurrencyId::Native,
+					),
+					None
+			)
+	}
+
+	  purge_transfer_allowance_remaining_metadata {
+			  let (sender, receiver) = set_up_users::<T>();
+		let receiver_1 = set_up_second_reciever::<T>();
+			  Pallet::<T>::add_transfer_allowance(RawOrigin::Signed(sender.clone()).into(), CurrencyId::Native, receiver.clone().into())?;
+			  Pallet::<T>::add_transfer_allowance(RawOrigin::Signed(sender.clone()).into(), CurrencyId::Native, receiver_1.clone().into())?;
+			  Pallet::<T>::remove_transfer_allowance(RawOrigin::Signed(sender.clone()).into(), CurrencyId::Native, receiver.clone().into())?;
+	  }:purge_transfer_allowance(RawOrigin::Signed(sender.clone()), CurrencyId::Native, receiver.clone().into())
+		  verify {
```

Comment:
A comment about `verify` in macros. The intention of this part of the macro is not to check the behavior itself (this is done well in the tests); it's to check that you're testing the worst case for that extrinsic. From my understanding of the library, if the extrinsic went ok, it computes the more expensive path, so in your case, you don't need to fill `verify` sections. So you can remove this safety and have less code to maintain (This also applies to other benchmarks)

Reply:
IMO, there should be at least one check still in `verify`. It is true that is a bit redundant with unit tests. However, the latter only work against a mock whereas real benchmarks apply the actual runtime configuration.

Reply:
The system automatically verifies that it returns `Ok()`, so there is already an implicit "verification". In most cases, you don't need to go into more detail here. Although, it all depends on the extrinsic complexity of course.